charts/crossplane-keycloak-realm/templates/bootstrap-realm-admin.yaml

```diff
@@ -0,0 +1,92 @@
+{{- if .Values.adminUser.enabled }}
+{{- $providerConfigName := tpl .Values.providerConfig.name $ | required ".Values.providerConfig.name is required" }}
+{{- $providerConfigKind := tpl .Values.providerConfig.kind $ | required ".Values.providerConfig.kind is required" }}
+{{- $providerConfigSecretName := tpl .Values.providerConfig.secretName $ | required ".Values.providerConfig.secretName is required" }}
+{{- $username := .Values.adminUser.username }}
+{{- $email := .Values.adminUser.email }}
```

Comment:
In general, prefer to reference `.Values.whatever.key` directly. Where a value is used multiple times, and there's some level of complexity (or at least logic), then I'm in favor of defining a temp variable. Please drop `$username`, `$email`, `$namespace`, and `$userRoleResourceName`, .